### PR TITLE
[PWA-1020] Upgrade Node 14 and Scaffold Node 16

### PR DESCRIPTION
**Pull request by mikhaelbois**
## Description

Upgrade Node version  minimum requirement form Node 10 to Node 14.
Fix issues with NPM 7 and above.

## Related Issue

Closes https://jira.corp.magento.com/browse/PWA-1020.

#### Test scenario(s) for direct fix/feature

- Verify scaffold build runs Node 14 (Node 16 is not yet supported on AWS)
- Verify node 14 is by default across all builds

Please locally test on both [Node 14 (Npm 6)](https://nodejs.org/download/release/v14.18.1/) and [Node 16 (Npm 8)](https://nodejs.org/en/).

You can run local scaffolding installation with this command:
`DEBUG_PROJECT_CREATION=1 ./packages/pwa-buildpack/bin/buildpack create-project scaffold-test --template "packages/venia-concept" --name "scaffold-test" --author "Scaffold Test" --backend-url "https://master-7rqtwti-mfwmkrjfqvbjk.us-4.magentosite.cloud/" --npm-client "npm"`

#### Test scenario(s) for any existing impacted features/areas

- Do and overall QA of the website and all critical features.
- Run Storybook.
- Run Cypress tests.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: prod.dockerfile
Comment (dpatil-magento) [inline prod.dockerfile:1]: AWS still does not have node 16 support, please change it to 14

Comment (mikhaelbois) [reply prod.dockerfile]: Done ✅

**Comment by dpatil-magento** [comment]
CI jobs are being verified in QA env, we will update this before it gets merged.

**Comment by michaelyu0123** [comment]
@mikhaelbois are we sure on removing the peerDependencies?

**Comment by mikhaelbois** [comment]
@michaelyu0123 
Reapplied `peerDependencies` and fixed merge conflicts.

**Comment by michaelyu0123** [comment]
Pr and scaffold tests are failing due to cicd not updated yet. LGTM Will need to revisit after QA and before merge.

File: packages/babel-preset-peregrine/package.json
Comment (dpatil-magento) [inline packages/babel-preset-peregrine/package.json:3]: @mikhaelbois Can all of these @magento/<package> (internal packages) be reverted back?  During release alpha package we update all needed package versions at once.

Comment (mikhaelbois) [reply packages/babel-preset-peregrine/package.json:3]: We can, but some were forgotten last time and that's what caused issues with NPM 7+ because of the peer dependencies that are more strict.
I will revert them all and do a reminder when a release 13.

Comment (dpatil-magento) [reply packages/babel-preset-peregrine/package.json:3]: Thanks @mikhaelbois . I will check this too. We cant continue with this as our release script does not have capability to identify any in-between release changes.

Comment (mikhaelbois) [reply packages/babel-preset-peregrine/package.json]: @dpatil-magento I just pushed the revert.

**Comment by dpatil-magento** [comment]
QA Approved. I will sync with @sharkySharks for CI builds and merge this.

**Review by dpatil-magento — APPROVED** [review]

**Review by sharkySharks — APPROVED** [review]

**Pull request merged**
